tools/mmp/driver.cs

```diff
@@ -800,12 +834,54 @@ static void Pack (IList<string> unprocessed)
 			if (App.LinkMode != LinkMode.All && App.RuntimeOptions != null)
 				App.RuntimeOptions.Write (App.AppDirectory);
 
+			if (aot != AotType.None) {
+				if (!IsUnified)
+					throw new MonoMacException (98, true, "AOT compilation is only available on Unified", ret);
+
+				AOTBundle ();
+				Watch ("AOT Compile", 1);
+			}
+
 			if (!string.IsNullOrEmpty (certificate_name)) {
 				CodeSign ();
 				Watch ("Code Sign", 1);
 			}
 		}
 
+		static void AOTBundle ()
+		{
+			string monoExe = $"/Library/Frameworks/Mono.framework/Commands/mono" + (Is64Bit ? "64" : "32");
+			var dirInfo = new DirectoryInfo (mmp_dir);
+
+			Func <FileInfo, bool> aotableFile = x => {
+				string extension = x.Extension.ToLowerInvariant ();
+				if (extension != ".exe" && extension != ".dll")
+					return false;
+
+				Console.WriteLine (x.Name);
+				if (aot_ignore.Contains (x.Name))
+					return false; 
+				if (aot_ignore.Count > 0)
+					Console.WriteLine (aot_ignore[0]);
+
+				switch (aot) {
+				case AotType.All:
+					return true;
+				case AotType.Platform:
+					return x.Name == "Xamarin.Mac.dll" || x.Name == "System.dll" || x.Name == "mscorlib.dll";
+				case AotType.Explicit:
+					return aot_explicit_reference.Contains (x.Name);
+				default:
+					throw new InvalidOperationException ($"AOTBundle with aot: {aot}?");
+				}
+			};
+
+			Parallel.ForEach (dirInfo.EnumerateFiles ().Where (aotableFile), new ParallelOptions () { MaxDegreeOfParallelism = Driver.Concurrency }, file => {
+				if (RunCommand (monoExe, String.Format ("--aot=hybrid {0}", Quote (file.ToString ())), new string [] {"MONO_PATH", mmp_dir }) > 0)
+					ErrorHelper.Warning (1337, string.Format ("AOT failed on {0}", file));
```

Comment:
Maybe make this an error? Users can now ignore specific assemblies if they're causing trouble, while still AOT-ing the rest, so making it an error won't prevent people from enabling AOT.